mmcv/transform/processing.py

```diff
@@ -0,0 +1,382 @@
+        keep_channel (bool, optional): Whether keep channel number the same as
+            input. Defaults to False.
+        channel_weights (tuple, optional): Channel weights to compute gray
+            image. Defaults to (1., 1., 1.).
+        color_format (str, optional): Color format set to be any of 'bgr',
+            'rgb', 'hsv' . Defaults to 'bgr'.
+    """
+
+    def __init__(self,
+                 prob: float = 0.1,
+                 keep_channel: bool = False,
+                 channel_weights: Sequence[float] = (1., 1., 1.),
+                 color_format: str = 'bgr'):
+        super().__init__()
+        self.prob = prob
+        self.keep_channel = keep_channel
+        self.channel_weights = channel_weights
+        assert color_format in ['bgr', 'rgb', 'hsv']
+        self.color_format = color_format
+
+    def transform(self, results: dict) -> dict:
+        """Apply random grayscale on results.
+
+        Args:
+            results (dict): Result dict contains the data to transform.
+
+        Returns:
+           dict[str: list]: The augmented data, where each value is wrapped
+               into a list.
+        """
+        img = results['img']
+        # convert hsv to bgr
+        if self.color_format == 'hsv':
+            img = mmcv.hsv2bgr(img)
+        num_output_channels = img.shape[2]
+        if random.random() < self.prob:
+            if num_output_channels > 1:
+                assert num_output_channels == len(self.channel_weights)
+                normalized_weights = np.array(self.channel_weights) / sum(
+                    self.channel_weights)
+                img = sum([
+                    weight * img[:, :, i]
+                    for i, weight in enumerate(normalized_weights)
+                ])
+                if self.keep_channel:
+                    img = img[:, :, None]
+                    results['img'] = np.dstack(
+                        [img for _ in range(num_output_channels)])
+                else:
+                    results['img'] = img
+                return results
+        results['img'] = img
+        return results
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__
+        repr_str += f', prob = {self.prob}'
+        repr_str += f', keep_channel = {self.keep_channel}'
+        repr_str += f', channel_weights = {self.channel_weights}'
+        repr_str += f', color_format = {self.color_format}'
+        return repr_str
+
+
+@TRANSFORMS.register_module()
+class MultiScaleFlipAug(BaseTransform):
+    """Test-time augmentation with multiple scales and flipping.
+
+    An example configuration is as followed:
+
+    .. code-block::
+
+        img_scale=[(1333, 400), (1333, 800)],
+        flip=True,
+        transforms=[
+            dict(type='Normalize', **img_norm_cfg),
+            dict(type='Pad', size_divisor=32),
+            dict(type='ImageToTensor', keys=['img']),
+            dict(type='Collect', keys=['img']),
+        ]
```

Comment:
there is not `MultiScaleFlipAug` in example

Reply:
Provide a detailed example.